### PR TITLE
feat(lib/contracts): use create3 for proxy admin deployments

### PR DESCRIPTION
**Pull request by kevinhalliday**
Deploy ProxyAdmin via Create3 factory

task: none

### CODE REVIEW

File: lib/contracts/create3/utils.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package create3
+
+import (
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+// HashSalt returns the [32]byte hash of the salt string.
+func HashSalt(s string) [32]byte {
```
Comment (ttarsi) [inline lib/contracts/create3/utils.go:8]: maybe a test for this? might be overkill

**Review by ttarsi — APPROVED** [review]

File: lib/contracts/create3/utils.go (hunk 1 above)
Comment (kevinhalliday) [reply lib/contracts/create3/utils.go:8]: Do you know any good keccak hashes I could use

Comment (pavelnikolov) [reply lib/contracts/create3/utils.go:8]: I don't think we need to test the Go `crypto` package. However, we might have a simple test to verify if the hashing algorithm(s) have changed. E.g. use one well known input and compare against its well known output as a sanity check.

**Review by pavelnikolov — APPROVED** [review]

File: lib/contracts/create3/utils.go (hunk 1 above)
Comment (kevinhalliday) [reply lib/contracts/create3/utils.go:8]: found one

Comment (kevinhalliday) [reply lib/contracts/create3/utils.go:8]: yeah did that

**Pull request merged**